### PR TITLE
feat(gsoc'24): ts + vue integration and jquery removal in layout files

### PR DESCRIPTION
**Pull request by niladrix719**
Fixes [#109](https://github.com/CircuitVerse/upptime/issues/109)

#### Describe the changes you have made in this PR -

ts + vue integration and jquery removal in:

- layoutMode.ts
- layoutNode.ts
- layoutBuffer.ts

and created a layoutStore to manage the state

### CODE REVIEW

**Review by JoshVarga — APPROVED** [review]

File: src/components/Extra.vue [hunk 1]
```diff
@@ -320,4 +321,16 @@ import CustomShortcut from './DialogBox/CustomShortcut.vue'
 import InsertSubcircuit from './DialogBox/InsertSubcircuit.vue'
 import OpenOffline from './DialogBox/OpenOffline.vue'
 import ReportIssue from './ReportIssue/ReportIssue.vue'
+import { useLayoutStore } from '#/store/layoutStore'
+import { onMounted, ref } from 'vue'
+
+const layoutStore = useLayoutStore()
+
+const testbenchPanel = ref<HTMLElement | null>(null);
+const layoutElementPanel = ref<HTMLElement | null>(null);
```
Comment (vedant-jain03) [inline src/components/Extra.vue:330]:
```suggestion
const testbenchPanelRef = ref<HTMLElement | null>(null);
const layoutElementPanelRef = ref<HTMLElement | null>(null);
```

Comment (vedant-jain03) [reply src/components/Extra.vue:330]: better naming

File: src/components/Panels/ElementsPanel/ElementsPanel.vue [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <template>
     <div
-        ref="ElementsPanel"
+        ref="elementsPanel"
```
Comment (vedant-jain03) [inline src/components/Panels/ElementsPanel/ElementsPanel.vue:3]: Above comment

File: src/components/Panels/ElementsPanel/ElementsPanel.vue [hunk 3]
```diff
@@ -145,10 +145,14 @@ import metadata from '#/simulator/src/metadata.json'
...
 import modules from '#/simulator/src/modules'
-import { onBeforeMount, ref } from 'vue'
+import { onBeforeMount, onMounted, ref } from 'vue'
+import { useLayoutStore } from '#/store/layoutStore'
 var panelData = []
 window.elementHierarchy = metadata.elementHierarchy
 window.elementPanelList = []
+const layoutStore = useLayoutStore()
+
+const elementsPanel = ref<HTMLElement | null>(null);
```
Comment (vedant-jain03) [inline src/components/Panels/ElementsPanel/ElementsPanel.vue:155]: better naming

File: src/components/Panels/PropertiesPanel/LayoutProperty/LayoutProperty.vue [hunk 4]
```diff
@@ -112,12 +112,19 @@
...
-import { ref, watch } from 'vue'
+import { onMounted, ref, watch } from 'vue'
 import { tempBuffer, layoutFunctions } from '#/simulator/src/layoutMode'
 import { scheduleUpdate } from '#/simulator/src/engine'
 import PanelHeader from '#/components/Panels/Shared/PanelHeader.vue'
+import { useLayoutStore } from '#/store/layoutStore'
 
 const titleEnable = ref(tempBuffer.layout.titleEnabled)
+const layoutStore = useLayoutStore()
+const layoutDialog = ref<HTMLElement | null>(null);
```
Comment (vedant-jain03) [inline src/components/Panels/PropertiesPanel/LayoutProperty/LayoutProperty.vue:123]: better naming

File: src/components/Panels/TimingDiagramPanel/TimingDiagramPanel.vue [hunk 5]
```diff
@@ -66,6 +67,12 @@ const plotArea: PlotArea = _plotArea
 const buttons = ref<TimingDiagramButton[]>(buttonsJSON)
 const plotRef = ref<HTMLElement | null>(null)
 const cycleUnits = ref(1000)
+const timingDiagramPanel = ref<HTMLElement | null>(null);
```
Comment (vedant-jain03) [inline src/components/Panels/TimingDiagramPanel/TimingDiagramPanel.vue:70]: better naming

File: src/store/layoutStore.ts [hunk 6]
```diff
@@ -0,0 +1,84 @@
+import { defineStore } from "pinia";
+import { ref, Ref, watch } from "vue";
+
+export const useLayoutStore = defineStore("layoutStore", () => {
+  const layoutMode = ref(false);
+  const layoutDialog: Ref<HTMLElement | null> = ref(null);
+  const layoutElementPanel: Ref<HTMLElement | null> = ref(null);
+  const elementsPanel: Ref<HTMLElement | null> = ref(null);
+  const timingDiagramPanel: Ref<HTMLElement | null> = ref(null);
+  const testbenchPanel: Ref<HTMLElement | null> = ref(null);
```
Comment (vedant-jain03) [inline src/store/layoutStore.ts:10]: better naming

**Review by vedant-jain03 — APPROVED** [review]
Small comments, LGTM

**Pull request merged**